sdp/processors/base_processor.py

```diff
@@ -242,3 +260,21 @@ def finalize(self, metrics: List):
         logger.info("Total number of entries after processing: %d", self.number_of_entries)
         if self.total_duration != 0:
             logger.info("Total audio duration (hours) after processing: %.2f", self.total_duration / 3600)
+
+
+    def test(self):
+        """Applies processing to "test_cases" and raises an error in case of mismatch."""
+        for test_case in self.test_cases:
+            generated_outputs = self.process_dataset_entry(test_case["input"].copy())
+            # can only return 1 or zero entries
```

Comment:
Is this a required assumption? I think we can just allow test_cast['output'] to be a list and would be able to test 1->many case?